src/mlpack/tests/linear_regression_test.cpp

```diff
@@ -21,25 +21,30 @@ using namespace mlpack;
  * Creates two 10x3 random matrices and one 10x1 "results" matrix.
  * Finds B in y=BX with one matrix, then predicts against the other.
  */
-TEST_CASE("LinearRegressionTestCase", "[LinearRegressionTest]")
+TEMPLATE_TEST_CASE("LinearRegressionTestCase", "[LinearRegressionTest]",
+    arma::fmat, arma::mat)
 {
+  typedef TestType MatType;
+  typedef arma::Row<typename MatType::elem_type> RowType;
+  typedef arma::Col<typename MatType::elem_type> ColType;
+
   // Predictors and points are 10x3 matrices.
-  arma::mat predictors(3, 10);
-  arma::mat points(3, 10);
+  MatType predictors(3, 10);
+  MatType points(3, 10);
 
   // Responses is the "correct" value for each point in predictors and points.
-  arma::rowvec responses(10);
+  RowType responses(10);
 
   // The values we get back when we predict for points.
-  arma::rowvec predictions(10);
+  RowType predictions(10);
 
   // We'll randomly select some coefficients for the linear response.
-  arma::vec coeffs;
+  ColType coeffs;
```

Comment:
We did not follow the same logic for these with the lars testing, but I think it is fine, we are not testing with Bandicoot in any case.

Also there is no reason for a test to pass with armadillo and fail with bandicoot

Reply:
It could definitely happen if there is a bug in Bandicoot 😄 we'll find out as we go. 👍